pkg/ring/tokens.go

```diff
@@ -0,0 +1,80 @@
+package ring
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+)
+
+// Tokens is a simple list of tokens.
+type Tokens []uint32
+
+// StoreToFile stores the tokens in the given directory.
+func (t Tokens) StoreToFile(tokenFilePath string) error {
+	if tokenFilePath == "" {
+		return errors.New("path is empty")
+	}
+
+	// If any operations failed further in the function, we keep the temporary
+	// file hanging around for debugging.
+	f, err := ioutil.TempFile(os.TempDir(), "tokens")
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		// If the file was not closed, then there must already be an error, hence ignore
+		// the error (if any) from f.Close(). If the file was already closed, then
+		// we would ignore the error in that case too.
+		f.Close()
```

Comment:
Please use `_ = f.Close()` to silence checkers that verify unused returned error.